src/output-json-file.c

```diff
@@ -174,84 +174,8 @@ JsonBuilder *JsonBuildFileInfoRecord(const Packet *p, const File *ff,
 
     jb_set_string(js, "app_proto", AppProtoToString(p->flow->alproto));
 
-    /* Open the fileinfo object. */
     jb_open_object(js, "fileinfo");
-
-    size_t filename_size = ff->name_len * 2 + 1;
-    char filename_string[filename_size];
-    BytesToStringBuffer(ff->name, ff->name_len, filename_string, filename_size);
-    jb_set_string(js, "filename", filename_string);
-
-    jb_open_array(js, "sid");
-    for (uint32_t i = 0; ff->sid != NULL && i < ff->sid_cnt; i++) {
-        jb_append_uint(js, ff->sid[i]);
-    }
-    jb_close(js);
-
-#ifdef HAVE_MAGIC
-    if (ff->magic)
-        jb_set_string(js, "magic", (char *)ff->magic);
-#endif
-    jb_set_bool(js, "gaps", ff->flags & FILE_HAS_GAPS);
-    switch (ff->state) {
-        case FILE_STATE_CLOSED:
-            jb_set_string(js, "state", "CLOSED");
-#ifdef HAVE_NSS
-            if (ff->flags & FILE_MD5) {
-                size_t x;
-                int i;
-                char str[256];
-                for (i = 0, x = 0; x < sizeof(ff->md5); x++) {
-                    i += snprintf(&str[i], 255-i, "%02x", ff->md5[x]);
-                }
-                jb_set_string(js, "md5", str);
-            }
-            if (ff->flags & FILE_SHA1) {
-                size_t x;
-                int i;
-                char str[256];
-                for (i = 0, x = 0; x < sizeof(ff->sha1); x++) {
-                    i += snprintf(&str[i], 255-i, "%02x", ff->sha1[x]);
-                }
-                jb_set_string(js, "sha1", str);
-            }
-#endif
-            break;
-        case FILE_STATE_TRUNCATED:
-            jb_set_string(js, "state", "TRUNCATED");
-            break;
-        case FILE_STATE_ERROR:
-            jb_set_string(js, "state", "ERROR");
-            break;
-        default:
-            jb_set_string(js, "state", "UNKNOWN");
-            break;
-    }
-
-#ifdef HAVE_NSS
-    if (ff->flags & FILE_SHA256) {
-        size_t x;
-        int i;
-        char str[256];
-        for (i = 0, x = 0; x < sizeof(ff->sha256); x++) {
-            i += snprintf(&str[i], 255-i, "%02x", ff->sha256[x]);
-        }
-        jb_set_string(js, "sha256", str);
-    }
-#endif
-
-    jb_set_bool(js, "stored", stored ? true : false);
-    if (ff->flags & FILE_STORED) {
-        jb_set_uint(js, "file_id", ff->file_store_id);
-    }
-    jb_set_uint(js, "size", FileTrackedSize(ff));
-    if (ff->end > 0) {
-        jb_set_uint(js, "start", ff->start);
-        jb_set_uint(js, "end", ff->end);
-    }
-    jb_set_uint(js, "tx_id", ff->txid);
-
-    /* Close fileinfo object */
+    EveFileInfo(js, ff, stored);
```

Comment:
I wonder if all this code should stay in output-json-file.. Thats the more common pattern rather than moving it into output-json.c

Reply:
It's now used from multiple places. However, it can be moved back.